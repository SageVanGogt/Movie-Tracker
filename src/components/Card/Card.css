.card {
  position: relative;
  width: 250px;
  height: 408px;
  margin: 20px;
  padding: 0;
  box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.30);
  background-color: white;
}

.movie-img {
  height: 370px;
  background-size: cover;
  box-sizing: border-box;
  position: relative;
  z-index: 0;
}


.vote-avg-circle {
  width: 50px;
  height: 50px;
  border-radius: 100%;
  background-color: #a108ff;
  position: absolute;
  left: 180px;
  top: 20px;
  box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.40);
}

.avg-score {
  font-family: 'Fira Sans', sans-serif;
  color: white;
  font-weight: 600;
  font-size: 1.1rem;
  font-style: italic;
  letter-spacing: 2px;
  text-align: center;
  margin-top: 15px;
}

.movie-title {
  font-family: 'Fira Sans', sans-serif;
  color: white;
  font-weight: 600;
  font-size: 1.2rem;
  text-align: left;
  letter-spacing: 1px;
  margin: 0;
}

.movie-date {
  font-family: 'Fira Sans', sans-serif;
  color: white;
  font-weight: 400;
  font-style: italic;
  font-size: 0.9rem;
  text-align: left;
  margin: 0;
}

.favorite-button {
  position: absolute;
  bottom: 0;
  text-align: center;
  width: 250px;
  padding-top: 8px;
  background-color: #a108ff;
  color: #FFF;
  height: 30px;
  box-sizing: content-box;
  border: none;
  cursor: pointer;
}

.delete-button {
  position: absolute;
  bottom: 0;
  text-align: center;
  width: 246.8px;
  padding-top: 8px;
  background-color: whitesmoke;
  color: #a108ff;
  border: 2px solid #a108ff;
  height: 30px;
  box-sizing: content-box;
  /* border: none; */
  cursor: pointer;
}

.delete-button:hover, .favorite-button:hover {
  background-color: #b748fc;
}

.delete-button:hover {
  color: white;
}

.overview {
  opacity: 0;
  position: absolute;
  width: 250px;
  height: 370px;
  color: #fff;
  background-color: rgba(17, 0, 29, 0.675);
  z-index: 1;
  font-family: 'Fira Sans', sans-serif;
  font-weight: 300;
  overflow: scroll;
  padding: 30px;
  box-sizing: border-box;
  text-align: left;
  font-size: 0.8rem;
  line-height: 1.2rem;
  transition: opacity 1s ease-out;
}

.overview:hover {
  opacity: 1;
}
