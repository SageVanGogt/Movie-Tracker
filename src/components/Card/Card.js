import React from "react";
import { postFavoriteToDb, removeFavorite } from './../../apiCall/apiCall';
import { connect } from 'react-redux';
import {removeFavoriteFromStore} from '../../actions/index';

export const Card = props => {
  const {
    movie_id,
    vote_average,
    title,
    poster_path,
    overview,
    release_date
  } = props;
  
  const handleDeleteClick = () => {
    removeFavorite({
          user_id: props.user.user_id,
          movie_id
        })
    props.handleRemoveFavorite(movie_id)
      
  }

  return (
    <div>
      <div>{vote_average}</div>
      <img src={poster_path}/>
      <h2>{title}</h2>
      <h4>{release_date}</h4>
      <div className="overview">
        <p>{overview}</p>
      </div>
      <div type="button" onClick={() => postFavoriteToDb(props, props.user)}>Favorite</div>
      <div type = "button"
      onClick = {
        () => handleDeleteClick()
      } > Remove Favorite </div>
    </div>
  );
};

export const mapStateToProps = (state) => ({
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  handleRemoveFavorite: (movieId) => dispatch(removeFavoriteFromStore(movieId))
})

export default connect(mapStateToProps, mapDispatchToProps)(Card);
