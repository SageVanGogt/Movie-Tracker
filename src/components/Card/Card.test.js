import React from 'react';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps, Card } from './Card';

describe('Card', () => {
  it('should match the snapshot', () => {
    let mockProps = {
        film_id: 1,
        vote_average: 34,
        title: "Fury",
        poster_path: "www.theworstmovieever.com",
        overview: "a bunch of text about how bad this move is",
        release_date: 2014
      }
    let wrapper = shallow(<Card {...mockProps}/>)
  
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should pull the correct props from the store', () => {
      let mockState = {
        user: {id: 1, name: 'lisa'},
        films: []
      }
      let mappedProps = mapStateToProps(mockState);
      let expected = {user: {id: 1, name: 'lisa'}};

      expect(mappedProps).toEqual(expected);
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct params', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch)
      const mockAction = {type: 'REMOVE_FAVORITE', movie_id: 45}
      const mockArg = 45

      mappedProps.handleRemoveFavorite(mockArg)

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    })
  })
});