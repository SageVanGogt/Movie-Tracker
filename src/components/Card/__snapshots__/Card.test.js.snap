// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    film_id={1}
    overview="a bunch of text about how bad this move is"
    poster_path="www.theworstmovieever.com"
    release_date={2014}
    title="Fury"
    vote_average={34}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          34
</div>,
        <img
          src="www.theworstmovieever.com"
/>,
        <h2>
          Fury
</h2>,
        <h4>
          2014
</h4>,
        <div
          className="overview"
>
          <p>
                    a bunch of text about how bad this move is
          </p>
</div>,
        <div
          onClick={[Function]}
          type="button"
>
          Favorite
</div>,
        <div
          onClick={[Function]}
          type="button"
>
           Remove Favorite 
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 34,
        },
        "ref": null,
        "rendered": 34,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "www.theworstmovieever.com",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Fury",
        },
        "ref": null,
        "rendered": "Fury",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 2014,
        },
        "ref": null,
        "rendered": 2014,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            a bunch of text about how bad this move is
</p>,
          "className": "overview",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "a bunch of text about how bad this move is",
          },
          "ref": null,
          "rendered": "a bunch of text about how bad this move is",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Remove Favorite ",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": " Remove Favorite ",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            34
</div>,
          <img
            src="www.theworstmovieever.com"
/>,
          <h2>
            Fury
</h2>,
          <h4>
            2014
</h4>,
          <div
            className="overview"
>
            <p>
                        a bunch of text about how bad this move is
            </p>
</div>,
          <div
            onClick={[Function]}
            type="button"
>
            Favorite
</div>,
          <div
            onClick={[Function]}
            type="button"
>
             Remove Favorite 
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 34,
          },
          "ref": null,
          "rendered": 34,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "www.theworstmovieever.com",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fury",
          },
          "ref": null,
          "rendered": "Fury",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 2014,
          },
          "ref": null,
          "rendered": 2014,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              a bunch of text about how bad this move is
</p>,
            "className": "overview",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "a bunch of text about how bad this move is",
            },
            "ref": null,
            "rendered": "a bunch of text about how bad this move is",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Remove Favorite ",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": " Remove Favorite ",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
